src/core_modules/capture-core/components/WidgetsChangelog/common/ChangelogFilterBar/ChangelogFilterBar.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React, { useMemo, useState, useCallback } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { withStyles } from '@material-ui/core/styles';
+import { spacers } from '@dhis2/ui';
+import type { ChangelogFilterProps, FilterValueType } from './ChangelogFilter.types';
+import { DropdownFilter } from './DropdownFilter';
+import { FIELD_TYPES, FILTER_TARGETS, CHANGELOG_ENTITY_TYPES } from '../Changelog/Changelog.constants';
+
+const styles = {
+    container: {
+        padding: `0 0 ${spacers.dp8} 0`,
+    },
+};
+
+const getFilterField = (id: string, entityType: string) => {
+    if (entityType === CHANGELOG_ENTITY_TYPES.TRACKED_ENTITY) {
+        return FILTER_TARGETS.ATTRIBUTE;
+    }
+    if (Object.values(FIELD_TYPES).includes(id)) {
+        return FILTER_TARGETS.FIELD;
+    }
+    return FILTER_TARGETS.DATA_ELEMENT;
+};
+
+const ChangelogFilterBarPlain = ({
+    classes,
+    filterValue,
+    setFilterValue,
+    attributeToFilterBy,
+    setAttributeToFilterBy,
+    dataItemDefinitions,
+    entityType,
+    entityData,
+}: ChangelogFilterProps) => {
+    const [openMenu, setOpenMenu] = useState <string | null>(null);
+
+    const toggleMenu = useCallback((menuName: string) => {
+        setOpenMenu(prev => (prev === menuName ? null : menuName));
+    }, []);
+
+    const handleItemSelected = useCallback(
+        (value: FilterValueType) => {
+            if (value === null) {
+                setFilterValue(null);
+                setAttributeToFilterBy(null);
+            } else {
+                const filterField = getFilterField(value.id, entityType);
+                setFilterValue(value);
+                setAttributeToFilterBy(filterField);
+            }
+        },
+        [setFilterValue, setAttributeToFilterBy, entityType],
+    );
+
+    const filterItems = useMemo(() =>
+        (entityType === CHANGELOG_ENTITY_TYPES.TRACKED_ENTITY
+            ? Object.keys(entityData).map((key) => {
+                const attribute = entityData[key];
+                return {
+                    id: key,
+                    name: attribute.name,
+                };
+            })
+            : Object.keys(dataItemDefinitions).map(key => ({
+                id: key,
+                name: dataItemDefinitions[key].name,
+            }))),
+    [entityType, dataItemDefinitions, entityData],
+    );
```

Comment:
Hi @henrikmv, I think that TET geometry should not be inside `dataItemDefinitions`, so there would be no need to filter it here. My suggestion would be that when creating [dataItemDefinitions](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetProfile/OverflowMenu/TrackedEntityChangelogWrapper/TrackedEntityChangelogWrapper.component.js#L21) in TrackedEntityChangelogWrapper you add only the attributes. Something like this should work:

```
const fieldElementsById = elements.reduce((acc, element: DataElement) => {
  if (!transformedTrackedEntityData[element.id]) {
    return acc;
  }
  ....
```
WDYT? Thanks!